roles/ircbouncer/tasks/znc.yml

```diff
@@ -42,9 +42,13 @@
 - name: Create znc config directory
   file: state=directory path=/usr/lib/znc/configs group=znc owner=znc
 
+- name: Notify user that irc_nick must be defined
+  debug: msg="`irc_nick` and related variables must be defined for znc to work."
+  when: irc_nick is not defined
+
 - name: Copy znc configuration file into place
   template: src=usr_lib_znc_configs_znc.conf.j2 dest=/usr/lib/znc/configs/znc.conf owner=znc group=znc
-  when: znc_config.rc != 0
+  when: znc_config.rc != 0 and irc_nick is defined
```

Comment:
It seems odd that you are trying to cover a misconfiguration error here when we don't in other places, e.g., prosody setup and mailserver setup.